yb-voyager/src/query/queryissue/parser_issue_detector.go

```diff
@@ -988,41 +943,49 @@ func (p *ParserIssueDetector) SetColumnStatistics(columnStats []utils.ColumnStat
 	}
 }
 
-// addIndexToCoverage adds an index and its prefixes to the coverage map for checking missing foreign key index issue
+// addIndexToCoverage adds an index to the table indexes map for checking missing foreign key index issue
 func (p *ParserIssueDetector) addIndexToCoverage(index *queryparser.Index) {
 	tableName := index.GetTableName()
 
-	// Extract columns in order (only non-expression columns)
-	columns := make([]string, 0)
-	for _, param := range index.Params {
-		if !param.IsExpression {
-			columns = append(columns, param.ColName)
+	// Add the index to the table's index list
+	if _, exists := p.tableIndexes[tableName]; !exists {
+		p.tableIndexes[tableName] = make([]*queryparser.Index, 0)
+	}
+	p.tableIndexes[tableName] = append(p.tableIndexes[tableName], index)
+}
+
+// addConstraintAsIndex creates a mock index object from a constraint and adds it to the table indexes map.
+// This is used for primary keys and unique constraints which are also indexes for missing foreign key index detection.
+func (p *ParserIssueDetector) addConstraintAsIndex(tableName string, columns []string, constraintType string) {
+	// Create mock index parameters from columns
+	indexParams := make([]queryparser.IndexParam, len(columns))
+	for i, colName := range columns {
+		indexParams[i] = queryparser.IndexParam{
+			ColName:      colName,
+			IsExpression: false,
 		}
 	}
 
-	// Generate all prefix combinations and store in coverage map
-	p.addIndexPrefixesToCoverageMap(tableName, columns)
-}
+	// Extract schema name from tableName (format: "schema.table")
+	schemaName := ""
+	tableNameOnly := tableName
```

Comment:
Instead of figuring that out here, you can pass in the proper schema name, table name from the caller. At the caller function, you have access to `queryparser.Table` type, which has both the fields